frontend/viewer/src/lib/components/ui/resizable/resizable-handle.svelte

```diff
@@ -3,28 +3,94 @@
   import type {WithoutChildrenOrChild} from 'bits-ui';
   import * as ResizablePrimitive from 'paneforge';
   import {Icon} from '../icon';
+  import Button, {type ButtonProps} from '../button/button.svelte';
+  import {Debounced} from 'runed';
+  import {type IconClass} from '$lib/icon-class';
+  import {onMount, tick} from 'svelte';
 
   let {
     ref = $bindable(null),
     class: className,
     withHandle = false,
+    leftPane,
+    rightPane,
     ...restProps
   }: WithoutChildrenOrChild<ResizablePrimitive.PaneResizerProps> & {
     withHandle?: boolean;
+    leftPane?: ResizablePrimitive.Pane;
+    rightPane?: ResizablePrimitive.Pane;
   } = $props();
+
+  let dragging = $state(false);
+  const draggingDebounced = new Debounced(() => dragging, 300);
+
+  type OnClickHandler = ButtonProps['onclick'];
+  const leftResizerOnClick = $derived.by(() => getPaneResizerOnClick(leftPane, rightPane));
+  const rightResizerOnClick = $derived.by(() => getPaneResizerOnClick(rightPane, leftPane));
+  function getPaneResizerOnClick(pane?: ResizablePrimitive.Pane, otherPane?: ResizablePrimitive.Pane): OnClickHandler {
+    if (!pane || !otherPane) return undefined;
+    if (pane.isCollapsed()) return undefined;
+    return () => {
+      if (otherPane.isCollapsed()) {
+        otherPane.expand();
+      } else {
+        pane.collapse();
+      }
+    };
+  }
+
+  let defaultPaneProportions = $state<[number, number]>();
+
+  onMount(async () => {
+    await tick(); // one of the panes almost definitely gets mounted after us
+    if (!leftPane || !rightPane) return;
+    const totalSize = leftPane.getSize() + rightPane.getSize();
+    defaultPaneProportions = [leftPane.getSize() / totalSize, rightPane.getSize() / totalSize];
```

Comment:
hum, so I think this means that it will reset to the proportions that the panes started with, so once we start persisting that size this would actually reset to the persisted size and not actually the default. I'm not sure that's what we want here.

Reply:
I added a `resetTo` prop to the component to allow configuring the dimensions that should be used.